### DISCUSSION TITLE
Limit number of search cycles in paperqa

### DISCUSSION THREAD

**Discussion by lmeder** [Future-House/paper-qa#764] [Q&A] [answered]
Hello everyone, 
I'm working with `paperqa` to extract information from texts trying to figure out topics, debates, paradigm shifts or contradictions in historical academic literature and it works very well! 
Although currently, when `paperqa` can't find any answers to a question, it seems to continue searching indefinitely. Since every additional search cycle incurs costs for me, it would be very helpful to have an option to limit the number of attempts or search cycles. Is there a way to adress this issue?

**Comment by jamesbraza** [comment, marked as answer]
There is `Settings.agent.max_timesteps` and `Settings.agent.timeout`: https://github.com/Future-House/paper-qa/blob/v5.8.0/paperqa/settings.py#L439

Does that work for you?

**Reply by lmeder** [reply]
Works fine, thank you!
